states/florida/main.py

```diff
@@ -1,24 +1,31 @@
-import json
-import glob
-import os
+from bs4 import BeautifulSoup
 
-from common import dir_path
+from common import dir_path, cache_request, decode_email, diff_and_save
 
+BASE_URL = 'https://dos.elections.myflorida.com/supervisors/'
 
-files = glob.glob(dir_path(__file__) + '/cache/*.json')
-data = []
-print(f'Found {len(files)} files')
-for file in files:
-  with open(file) as fh:
-    datum = json.load(fh)
-    county = datum['title'].split('Supervisor')[0].strip()
-    data += [{
-      'locale': county,
-      'official': datum['name'].replace(u'\xa0', ' ').split(',')[0].strip(),
-      'emails': [datum['email'].split(':')[1].strip()],  # ignore leading 'mailto:'
-      'url': datum['url'],
-      'county': county,
-    }]
 
-with open('public/florida.json', 'w') as fh:
-  json.dump(data, fh)
+def parse_county(soup):
+  county = soup.find('p', class_='title').text.split('Supervisor')[0].strip()
+  links = soup.find(id='rightContent')('a')
+  return {
+    'locale': county,
+    'official': soup.find('span', class_='bigRed').text.replace(u'\xa0', ' ').split(',')[0].strip(),
+    'emails': [decode_email(links[0].find('span', class_='__cf_email__').get('data-cfemail')).strip()],
+    'url': links[1]['href'].strip(),
+    'county': county,
+  }
+
+if __name__ == '__main__':
+  data = []
+  text = cache_request(BASE_URL)
+  soup = BeautifulSoup(text, 'html.parser')
+
+  county_links = map(lambda x: x['href'], soup.select('a[href^=countyInfo]'))
```

Comment:
Don't use map: instead, use a list comprehension: https://realpython.com/list-comprehension-python/

You could even merge this logic into the following for loop.

Reply:
Removed map (Florida).